eZ/Publish/Core/Repository/ContentService.php

```diff
@@ -965,8 +965,13 @@ protected function publishUrlAliasesForContent(APIContent $content, $updatePathI
     public function deleteContent(ContentInfo $contentInfo)
     {
         $contentInfo = $this->internalLoadContentInfo($contentInfo->id);
+        $content = $this->internalLoadContent($contentInfo->id);
```

Comment:
we don't need to load content, you can use contentInfo in all cases here afaik.

Reply:
+1

Reply:
hmm, you are right, `StatusLimitationType` needs it.

Two ways that can be solved:
- We fix `StatusLimitationType`, as we don't get VersionInfo from callee
- Change code here to skip loading of contentInfo, and use content all over.


For me I'm ok with both, on one side it's good that version based permissions asks for VersionInfo being given by callee, on the other side it makes it inconsistent with other content based limitations and in cases like this there is no value in loading it here as we don't know exact version anyway.

@alongosz / @adamwojs Any opinions? (also see coding style review note btw)